process_duty_email.py

```diff
@@ -0,0 +1,293 @@
+        f.write(text)
+
+def get_files():
+    """
+    This function gets the paths for csv files download by duty bioinformatician to the Duty_Bioinformatics_CSV folder.
+    """
+    folder = config.CSVread_folder
+    files = (
+        file
+        for file in os.listdir(folder)
+        if os.path.isfile(os.path.join(folder, file))
+    )
+    files_list = collections.defaultdict(list)
+    for filename in files:
+        results = re.search("__(\S+)__", filename)
+        project = results.group(1)
+        filepath = folder + "\\" + filename
+        files_list[project].append(filepath)
+    files_merged = dict(files_list)
+    return files_merged
+
+class Project:
+    """
+    Process project
+    """
+    def __init__(self, project, files):
+        self.project = project
+        self.project_files = files
+    def process_data(self):
+        """
+        This function executes the processing of a project based on the project type e.g. SNP, WES, TSO500 or MokaPipe
+        """
+        if self.project == "WES":
+            Project.process_WES(self)
+        elif self.project == "SNP":
+            Project.process_SNP(self)
+        elif self.project == "TSO500":
+            Project.process_TSO(self)
+        elif self.project == "MokaPipe":
+            Project.process_MokaPipe(self)
+    def process_WES(self):
+        """
+        Process WES Runs
+        """
+        destination_of_files = config.WES_destination_of_files
+        for file_path in self.project_files:
+            csv_file = pandas.read_csv(file_path, index_col=None)
+            url_wes = get_data(csv_file, '"{}"'.format(destination_of_files))
+            output = download_data(url_wes)
+            list_of_stings = file_path.split("\\")
+            save_log_file(output, list_of_stings[-1].replace('.csv','.txt'))
+            print(output)
+    def process_SNP(self):
+        """
+        Process SNP Runs
+        """
+        destination_of_VCFs = config.SNP_destination_of_VCFs
+        for file_path in self.project_files:
+            csv_file = pandas.read_csv(file_path, index_col=None)
+            url_snp = get_data(csv_file, '"{}"'.format(destination_of_VCFs))
+            output = download_data(url_snp)
+            list_of_stings = file_path.split("\\")
+            save_log_file(output, list_of_stings[-1].replace('.csv','.txt'))
+            print(output)
+    def process_MokaPipe(self):
+        """
+        Process MokaPipe Runs
+        """
+        # Path to St George's transfer folder
+        StG_transfer = config.StG_transfer
+        for file_path in self.project_files:
+            # asks for final part to the destination folder
+            output_folder = ask_for_folder()
+             # create directory for RPKM and download data
+            path_to_RPKM = output_folder + "\\RPKM"
+            destination_of_Coverage = output_folder + "\\coverage"
+            os.mkdir(path_to_RPKM)
+            os.mkdir(destination_of_Coverage)

```

Comment:
i assume this doesn't fail if the path already exists?

Reply:
it would fail if the path already exists.

Reply:
in which case it should go into a try/except statement - a library eg NGS500 can be split across multiple runs (eg NGS500A and NGS500B) which would go into the same folder on trust network (NGS500).